Sources/zkp/Combine.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  Combine.swift
+//  GigaBitcoin/secp256k1.swift
+//
+//  Copyright (c) 2023 GigaBitcoin LLC
+//  Distributed under the MIT software license
+//
+//  See the accompanying file LICENSE for information
+//
+
+import Foundation
+
+public extension secp256k1.Signing.PublicKey {
+    /// Create a new `PublicKey` by combining the current public key with an array of public keys.
+    /// - Parameters:
+    ///   - pubkeys: the array of public key objects to be combined with
+    ///   - format: the format of the combined `PublicKey` object
+    /// - Returns: combined `PublicKey` object
+    func combine(_ pubkeys: [Self], format: secp256k1.Format = .compressed) throws -> Self {
+        let context = secp256k1.Context.rawRepresentation
+        let allPubKeys = [self] + pubkeys
+        var publicKey = secp256k1_pubkey()
+        var pubKeyLen = format.length
+        var pubBytes = [UInt8](repeating: 0, count: pubKeyLen)
+
+        var keys = allPubKeys.map {
+            var newPubKey = secp256k1_pubkey()
+            $0.dataRepresentation.copyToUnsafeMutableBytes(of: &newPubKey.data)
+            let pointerKey: UnsafePointer<secp256k1_pubkey>? = withUnsafePointer(to: &newPubKey) { $0 }
+            return pointerKey
+        }
+
+        guard secp256k1_ec_pubkey_combine(context, &publicKey, &keys, pubkeys.count).boolValue,
```

Comment:
Is this supposed to be `pubkeys.count` or rather `allPubkeys.count`? Given that the function combines pubkeys + 1 (self)

Reply:
Good catch!

Reply:
Upon further review, now I'm not exactly sure which is correct from the documentation https://github.com/bitcoin-core/secp256k1/blob/1f1bb78b7fc2060d06df170d76d08511b077cfe7/include/secp256k1.h#L871